sgkit/stats/decomposition.py

```diff
@@ -0,0 +1,169 @@
+    * iterated_power : defaults to ``0``, the default for
+      ``dask.linalg.svd_compressed``.ad of numpy
+
+    Examples
+    --------
+    >>> from sgkit.stats.preprocessing import PattersonScaler
+    >>> from sgkit.stats.decomposition import GenotypePCA
+    >>> import dask.array as da
+
+    >>> # Let's generate some random diploid genotype data
+    >>> # With 30000 variants and 67 samples
+    >>> n_variants = 30000
+    >>> n_samples = 67
+    >>> genotypes = da.random.choice(3, n_variants * n_samples)
+    >>> genotypes = genotypes.reshape(n_variants, n_samples)
+
+    >>> scaler = PattersonScaler()
+    >>> scaled_genotypes = scaler.fit_transform(genotypes)
+    >>> # If you want to deal with the scaled values directly
+    >>> # scaled_genotypes.compute()
+    >>> # Or you can put the scaled_genotypes directly into the PCA
+    >>> pca = GenotypePCA()
+    >>> transformed = pca.fit_transform(scaled_genotypes)
+
+    >>> # Use SKLearn Pipelines
+    >>> # https://github.com/pystatgen/sgkit/issues/95#issuecomment-672879865
+    >>> from sklearn.pipeline import Pipeline
+    >>> est = Pipeline([ \
+        ('scaler', PattersonScaler()), \
+        ('pca', GenotypePCA(n_components=2)) \
+        ])
+    >>> pcs = est.fit_transform(genotypes)
+    >>> # `est` would also contain loadings + explained variance
+    >>> # `scaler` would contain the MAF and binomial variance values needed for out-of-sample projection
+    >>>   # Out-of-sample projection
+    >>> pcs_oos = est.transform(genotypes)
+    """
+
+    def __init__(
+        self,
+        n_components: int = 10,
+        copy: bool = True,
+        ploidy: int = 2,
+        iterated_power: int = 0,
+        random_state: Optional[int] = None,
+        svd_solver: str = "full",
+    ):
+        self.n_components = n_components
+        self.copy = copy
+        self.ploidy = ploidy
+        self.svd_solver = svd_solver
+        self.iterated_power = iterated_power
+        self.random_state = random_state
+
+    def fit(self, gn: ArrayLike, y: Optional[ArrayLike] = None) -> "GenotypePCA":
+        self._fit(gn)
+        return self
+
+    def _get_solver(self) -> str:
+        solvers = {"full", "randomized"}
+        solver = self.svd_solver
+
+        if solver not in solvers:
+            raise ValueError(
+                "Invalid solver '{}'. Must be one of {}".format(solver, solvers)
+            )
+        return solver
+
+    def fit_transform(self, gn: ArrayLike, y: Optional[ArrayLike] = None) -> ArrayLike:
+        u, s, v = self._fit(gn)
+        solver = self._get_solver()
+
+        if solver in {"full"}:
+            u = u[:, : self.n_components]
+            u *= s[: self.n_components]
+        else:
+            u *= s
+
+        return u
+
+    def _fit(self, gn: ArrayLike) -> Tuple[ArrayLike, ArrayLike, ArrayLike]:
+        x = gn.T
+        n_samples, n_features = x.shape
+
+        solver = self._get_solver()
```

Comment:
Wouldn't the centering be done in the scaler? Or am I missing something?

Reply:
It would and what I mean is dask-ml version is forcing the centering in case the user forgot to do it themselves (which is often ok in dask because it will recognize that tasks share upstream nodes in the graph).  If we're removing that safety measure though, we need to make it clear in the docs that the user is responsible for it.